Formula/nemu.rb

```diff
@@ -0,0 +1,30 @@
+class Nemu < Formula
+  desc "Ncurses UI for QEMU"
+  homepage "https://github.com/nemuTUI/nemu"
+  url "https://github.com/nemuTUI/nemu/archive/refs/tags/v3.2.0.tar.gz"
+  sha256 "f35b7e5c534bae294fe6df7685b686682145222a954b6914dfdcc17d386fecba"
+  license "BSD-2-Clause"
+
+  depends_on "cmake" => :build
+  depends_on "gettext" => :build
+  depends_on "pkg-config" => :build
+  depends_on "json-c"
+  depends_on "libarchive"
+  depends_on "ncurses"
+  depends_on "openssl@3"
+
+  on_linux do
+    depends_on "libusb"
+  end
+
+  def install
+    system "cmake", ".", "-DCMAKE_INSTALL_PREFIX=#{prefix}"
+    system "cmake", "--build", "."
+    system "make", "install"
+  end
+
+  test do
+    expected = "-s, --start"
+    assert_match expected, shell_output("#{bin}/nemu --help")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
For nemu to function, you need to generate a configuration file. I redid the tests based on this.

Reply:
On Ubuntu test is OK, but on MacOS it fauls with:
`nm_mkdir_parent: failed to create directory "/Users/brew/.config" (Operation not permitted)`

In CI on macos, applications are not allowed to create files in the home directory?

Reply:
This is strange, nemu uses $HOME/.config to generate path.
Ok, will try to fix this with XDG_CONFIG_HOME

Reply:
Now the `brew test nemu` pass, only warnings about `Dependency remain ... is provided by macOS`